staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -235,6 +235,21 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string at compilation time.
+	// If both Message and MessageExpression are present on a Rule, then MessageExpression will be used if validation
+	// fails. If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
+	// as if the MessageExpression field were unset.
+	// If string.format is used inside MessageExpression, then MessageExpression can also fail due to the number of
+	// arguments given to format() not matching the amount requested inside the format string, a type mismatch between
+	// an argument and a formatting clause (%s/%f/%d/etc.), and invalid syntax inside the formatting string itself. In any of those cases,
```

Comment:
Should all references to `string.format()` be dropped from this PR and added in the PR where `string.format()` is added? Doing so would keep the main branch in a consistent state.

cc @cici37

Reply:
Yeah, I didn't think about how the main branch would refer to non-existent functionality for a time - I'm in favor of removing it as well.

Reply:
+1 to keep it consistent

Reply:
Gotcha, fixed.